### PR TITLE
Added kafka consumer method

### PR DESCRIPTION
**Pull request by franciszekjob**

### CODE REVIEW

File: api/app.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+
+# user creates account
+@app.route("/users", methods=["POST"])
+def create_account(username, password):
+    return f"Hello f{username}"
+
+
+# user creates a bot
+@app.route("/<apiKey>/bots", methods=["POST"])
```
Comment (amasend) [inline api/app.py]: `api key` should not be send via path but rather inside request header

File: api/app.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+
+# user gets a current price
+@app.route("/current-prices", methods=["GET"])
+def current_prices():
+    return "Current prices"
+
+
+# user gets his trade history
+@app.route("/<apiKey>/trade-history", methods=["GET"])
```
Comment (amasend) [inline api/app.py]: same as above

File: history_writer/history_writer.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    bootstrap_servers = ['localhost:9092']
+
+    consumer = KafkaConsumer(
+        'crypto-trader-topic',
+        bootstrap_servers=bootstrap_servers,
+        auto_offset_reset='earliest',
+        enable_auto_commit=True)
+
+    for kafka_data in consumer:
+        data = kafka_data
```
Comment (amasend) [inline history_writer/history_writer.py]: Why we need another assignment here?

**Review by amasend — CHANGES_REQUESTED** [review]
PR has conflicts, please resolve them.
I see that API is mocked for now. We can accept this but in future this needs to have implementation.

File: history_writer/history_writer.py (hunk 3 above)
Comment (franciszekjob) [reply history_writer/history_writer.py]: Fixed now, didn't push changes to the repo

File: api/database/database_manager.py [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+        Paramaters
+        ------------
+        cursor: connection required - postresql connection
+        username: str required - username
+        password: str required - password
+        exchange_name: str required - name of and exchange, e.g. binance
+        symbol: str - symbol of cryptocurrency, e.g. BTC/USDT
+    """
+    user_data = authenticate_user(cursor, username, password)
+    if user_data != False:
```
Comment (amasend) [inline api/database/database_manager.py]: Why we have such weird comparison here?

```suggestion
    if not user_data:
```

File: api/database/database_manager.py [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+        try:
+            user_api_key = user_data[2]
+            cursor.execute("""INSERT INTO exchanges
+            VALUES('{0}', '{1}', '{2}', '{3}', '{4}');""".format(exchange_name, username, exchange_api_key,
+                                                                 user_api_key, secret))
+            connection.commit()
+            return "Exchange created"
+        except:
+            return "Error"
```
Comment (amasend) [inline api/database/database_manager.py]: It is always better to raise some exception when there is an error then returning a string.

File: api/database/database_manager.py [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+def create_user(connection, cursor, username, password, api_key):
+    """ Inserts username, password and an api key to the database
+    """
+    try:
+        cursor.execute("""INSERT INTO users (username,password, api_key)
+        VALUES(%s, %s, %s)
+        returning user;""", (username, password, api_key))
+        connection.commit()
+        return "User created"
+    except (Exception, psycopg2.DatabaseError) as error:
```
Comment (amasend) [inline api/database/database_manager.py:31]: Exception should be returned.

File: api/database/database_manager.py [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+    try:
+        user_data = cursor.fetchall()[0]
+        username = user_data[0]
+        db_password = user_data[1]  # hashed password from database
+        if username == username and sha256_crypt.verify(password, db_password):
+            # VERIFIED - username and password are correct
+            return user_data
+        else:
+            return "Username or password is incorrect"
+    except:
```
Comment (amasend) [inline api/database/database_manager.py:50]: Same as above.

File: api/database/database_manager.py [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+        exchange_data = cursor.fetchall()[0]
+        exchange_api_key = exchange_data[2]
+        secret = exchange_data[4]
+
+        exchange = getattr(ccxt, exchange_name)({'options': {'adjustForTimeDifference': True}})
+        exchange.apiKey = exchange_api_key
+        exchange.secret = secret
+        if exchange.has['fetchMyTrades']:
+            return exchange.fetch_my_trades(symbol=symbol)
+        return "Cannot get trades"
```
Comment (amasend) [inline api/database/database_manager.py]: Raise an exception.

File: api/database/database_manager.py [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+    else:
+        return "Username or password is incorrect"
```
Comment (amasend) [inline api/database/database_manager.py]: Same a above.

**Review by amasend — CHANGES_REQUESTED** [review]
Read about exception handling in Python.
It is always better to create own exceptions (meaningfull one) and implement a leayer that handles them eg in API to return appropriate errors to user.

**Review by amasend — CHANGES_REQUESTED** [review]
Please resolve conflicts and I will merge that PR.

**Review by amasend — APPROVED** [review]

**Pull request merged**